ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceBuilder.scala

```diff
@@ -252,11 +252,25 @@ private[platform] case class IndexServiceBuilder(
       headAtInitialization = (ledgerEnd.lastOffset, ledgerEnd.lastEventSeqId),
     )
 
+  private val APIStreamShutdownTimeout: FiniteDuration = FiniteDuration(5, "seconds")
+
   private def dispatcherOwner(ledgerEnd: Offset): ResourceOwner[Dispatcher[Offset]] =
     Dispatcher.owner(
       name = "sql-ledger",
       zeroIndex = Offset.beforeBegin,
       headAtInitialization = ledgerEnd,
+      releaseMapper = releaseFuture =>
+        Future.firstCompletedOf(
+          List(
+            releaseFuture,
+            akka.pattern.after(APIStreamShutdownTimeout, mat.system.scheduler) {
+              logger.warn(
+                s"Shutdown of API streams did not finish in ${APIStreamShutdownTimeout.toSeconds} seconds. System shutdown continues."
+              )
```

Comment:
Why does this block not run if the stream shutdown was successful? Is it because the release process shuts down the Akka system (`mat.system`), which cancels the task scheduled with `akka.pattern.after`?

Reply:
Yes, but good point, it is hardly "safe". In order to prevent misleading warning messages I added a timer-util Timeout to capture a more robust Future timeout approach, which is cancelable.